#!/usr/bin/env python2
PACKAGE = "gantry_crane_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("u_max", double_t, 0, "maximum control input",    200.0, 0.0,   1000.0)

controller_type_enum = gen.enum([ gen.const("NULL",       int_t, 0, "no control"),
                                  gen.const("PolePlacement", int_t, 1, "Zustandsregler (Polvorgabe Ackermann)"),
                                  gen.const("LQR",        int_t, 2, "Zustandsregler (LQR)"),
                                  gen.const("MPC",        int_t, 3, "Model Predictive Control")],
                                "Reglertyp")

gen.add("controller_type", int_t, 0, "", 1, 0, 3, edit_method=controller_type_enum)

exit(gen.generate(PACKAGE, "gantry_crane_controller", "Controller"))
